From 4279340e214d5af0574c415a36f6c512b8291665 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Fri, 22 Nov 2024 18:03:12 +0100
Subject: [PATCH 712/954] feat(stm32mp21): no Core1 reset on STM32MP21

C1P1 is not present on STM32MP21, don't manage its reset in that case.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 8af78baa7a4c202191982c8817ace480e03b798e feat(stm32mp21): no Core1 reset on STM32MP21]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: Ic991ebdd813d24aedc7c84e6dd4274d783ab7df5
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/420548
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/438133
Domain-Review: Maxime MERE <maxime.mere@foss.st.com>
Tested-by: Maxime MERE <maxime.mere@foss.st.com>
---
 plat/st/stm32mp2/bl2_plat_setup.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plat/st/stm32mp2/bl2_plat_setup.c b/plat/st/stm32mp2/bl2_plat_setup.c
index b6bb172f4b..ddad892b7a 100644
--- a/plat/st/stm32mp2/bl2_plat_setup.c
+++ b/plat/st/stm32mp2/bl2_plat_setup.c
@@ -157,12 +157,12 @@ static void print_reset_reason(void)
 		INFO("  IWDG5 system reset (rst_iwdg5)\n");
 		return;
 	}
-#endif /* !STM32MP21 */
 
 	if ((rstsr & RCC_C1BOOTRSTSCLRR_C1P1RSTF) != 0U) {
 		INFO("  A35 processor core 1 reset\n");
 		return;
 	}
+#endif /* !STM32MP21 */
 
 	if ((rstsr & RCC_C1BOOTRSTSCLRR_PADRSTF) != 0U) {
 		INFO("  Pad Reset from NRST\n");
-- 
2.53.0

